bluepysnap/edges.py

```diff
@@ -27,8 +26,353 @@
 from cached_property import cached_property
 
 from bluepysnap.exceptions import BluepySnapError
-from bluepysnap import utils
+from bluepysnap.circuit_ids import CircuitEdgeId, CircuitEdgeIds, CircuitNodeId, CircuitNodeIds
 from bluepysnap.sonata_constants import DYNAMICS_PREFIX, Edge, ConstContainer
+from bluepysnap import utils
+
+
+class Edges:
+    """The top level Edges accessor."""
+
+    def __init__(self, circuit):
+        """Initialize the top level Edges accessor."""
+        self._circuit = circuit
+        self._config = self._circuit.config['networks']['edges']
+        self._populations = self._get_populations()
+
+    def _get_populations(self):
+        """Collect the different EdgePopulation."""
+        res = {}
+        for file_config in self._config:
+            storage = EdgeStorage(file_config, self._circuit)
+            for population in storage.population_names:  # pylint: disable=not-an-iterable
+                if population in res:
+                    raise BluepySnapError("Duplicated edge population: '%s'" % population)
+                res[population] = storage.population(population)
+        return res
+
+    @cached_property
+    def population_names(self):
+        """Returns all the population names from the Circuit."""
+        return sorted(self._populations)
+
+    @cached_property
+    def property_dtypes(self):
+        """Returns all the property dtypes for the Circuit."""
+
+        def _update(d, index, value):
+            if index not in d:
+                d[index] = value
+            elif d[index] != value:
+                raise BluepySnapError("Same property with different "
+                                      "dtype. {}: {}!= {}".format(index, value, d[index]))
+
+        res = dict()
+        for pop in self.populations():
+            for varname, dtype in pop.property_dtypes.iteritems():
+                _update(res, varname, dtype)
+        return pd.Series(res)
+
+    def keys(self):
+        """Returns iterator on the population names.
+
+        Made to simulate the behavior of a dict.keys().
+        """
+        return (name for name in self.population_names)
+
+    def values(self):
+        """Returns iterator on the EdgePopulations.
+
+        Made to simulate the behavior of a dict.values().
+        """
+        return (self[name] for name in self.population_names)
+
+    def items(self):
+        """Returns iterator on the tuples (population name, EdgePopulations).
+
+        Made to simulate the behavior of a dict.items().
+        """
+        return ((name, self[name]) for name in self.population_names)
+
+    # helper renaming
+    names = keys
+    populations = values
```

Comment:
I'd put `names = keys` right after the definition of keys and same for populations but if you prefer to group them here, it's ok.

Reply:
Why these aliases at all?

Reply:
For readability in consumer codes. But I am not sure to keep them indeed and `names` is just a bad name.

I removed them.